ksqldb-engine/src/main/java/io/confluent/ksql/execution/pull/PullQueryResult.java

```diff
@@ -125,6 +126,9 @@ public void start() {
   public void stop() {
     try {
       pullQueryQueue.end();
+      pullQueryMetrics.ifPresent(pm -> pm.getCoordinatorThreadPoolGauge().update(
```

Comment:
Do you need to pass in to the metrics something coming from the metrics?  Can't you maybe just say pm.recordsThreadPoolSize()?  Also, I thought that usually the metrics system determines how often they poll their data.  Do you need to specifically record it on every request?